grpcutils/dial.go

```diff
@@ -59,8 +59,7 @@ func (o DialOpts) Dial(ctx context.Context) (*grpc.ClientConn, error) {
 		opts = append(opts, grpc.WithAuthority(o.Authority))
 	}
 	opts = append(opts, o.ExtraOptions...)
-	ctx, cancel := context.WithTimeout(ctx, time.Minute)
-	defer cancel()
+
```

Comment:
perhaps we should be returning cancel instead? plus the function signature change would make it obvious to devs about the change in behavior; so little risk of subtly breaking gloo edge @EItanya

Reply:
Does gloo edge crash when this function times out? @kdorosh

Reply:
Per https://github.com/solo-io/go-utils/pull/484#issuecomment-1168645638 i don't think it's used

but i do think returning the cancel is better so that the caller has control over short-circuiting the timeout (which is also ideally configurable). just for better code; not because we would break gloo edge

Reply:
Actually I think it's easier to just create a context specific to the grpc connection and cancel it from the calling function. So maybe open an issue to upgrade and add this instead of modifying the signature? Wdyt?

Reply:
@nfuden @EItanya @kdorosh I don't feel strongly about the solution here. But looks like the census wants to allow the duration to be configurable and pass back a cancelFn to the caller?

Reply:
That'd be breaking,so if we're gonna break I'd rather just create a new function with no opinionated timeout that we can use in GM.

Reply:
I'd argue its current behavior is confusing and borderline wrong, thus breaking has the added benefit of alerting users that behavior may not be what they expect

That said your approach Eitan is also very reasonable to me

Reply:
Hmm well if we want to keep the current behavior I can just close this PR no need to change it. I just noticed it while running the server with a debugger attached

Reply:
We're not saying always keep the same behavior, just have it be the default and allow overriding. But I also definitely see Kevin's point. I don't feel super strongly about it.

Reply:
left a review